### PR TITLE
Allow arbitrary pip providers

### PR DESCRIPTION
**Pull request by ethanhs**
#### Pull Request (PR) description
 This allows expressions such as `python3.6 -m pip` as a provider

#### This Pull Request (PR) fixes the following issues
n/a

### CODE REVIEW

File: manifests/pip.pp [hunk 1]
```diff
@@ -75,7 +75,7 @@
   $pkgname                             = $name,
   $ensure                              = present,
   $virtualenv                          = 'system',
-  Enum['pip', 'pip3'] $pip_provider    = 'pip',
+  $pip_provider                        = 'pip',
```
Comment (bastelfreak) [inline manifests/pip.pp]: can you use the datatype `String[1]` please?

**Comment by bastelfreak** [comment]
thanks for the PR! Can you please add an example to the README.md?

**Comment by danquack** [comment]
This relates to #462

**Comment by ethanhs** [comment]
Closing in favor of https://github.com/voxpupuli/puppet-python/pull/480

**Pull request closed without merging**